City-of-Helsinki/drupal-helfi-kasvatus-koulutus
# UHF-4092 Group news archive

# Group news archive [UHF-4092](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4092)
Created group news archive paragraph that can be added to landing pages.

## What was done
* Created paragraph group news archive that can be added to landing pages.
* Added link field for latest group news paragraph to link to the archive.
* Styled the paragraph.
* New display for news item called medium teaser was added.

## How to install
* Make sure your instance is up and running on latest dev branch with database synced from testing server.
    * `git pull origin dev`
    * `make fresh`
* Checkout this branch.
* Update the HDBT theme
    * `composer require drupal/hdbt:dev-UHF-4092_group_news_archive`
* Run `make drush-cim drush-cr`

## How to test
* Go to any of the groups and create news items under the group. Preferably create over 10 items but if you are feeling lazy you can just add a few and adjust the view to trigger the pager earlier so that you can make sure it works as well.
* Create a landing page under the group and to that landing page add group news archive paragraph. Pick the group that you are creating the archive for and save the page.
* Now go to edit for example the group front page and add latest group news paragraph there. It should now have also a field where you can link to the group news archive that you just created.
* Now on the page where you added the latest group news there should be a link "All news" that takes you to the group news archive and the archive should look similar to this layout: https://share.goabstract.com/8be8c801-48b8-40b5-866e-ec197360fccf?collectionLayerId=9ba478f1-16e6-4877-a836-4c3f296b69da&mode=design

## Other PRs
* https://github.com/City-of-Helsinki/drupal-hdbt/pull/297
* Here is a PR that is not part of this review but related to it: https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/314
